k8s/core/services.go

```diff
@@ -63,6 +67,30 @@ func (c *Client) GetService(svcName string, svcNS string) (*corev1.Service, erro
 	return c.core.Services(svcNS).Get(svcName, metav1.GetOptions{})
 }
 
+// ListServices list services using filters or list all if options are empty
+func (c *Client) ListServices(svcNamespace string, listOptions metav1.ListOptions) (*corev1.ServiceList, error) {
+	if err := c.initClient(); err != nil {
+		return nil, err
+	}
+
+	return c.core.Services(svcNamespace).List(listOptions)
+}
+
+// GetServiceEndpoint gets the externalIP if service is a LoadBalancer or ClusterIP otherwise
+func (c *Client) GetServiceEndpoint(svcName, namespace string) (string, error) {
+	if err := c.initClient(); err != nil {
+		return "", err
+	}
+	svc, err := c.GetService(svcName, namespace)
+	if err != nil {
+		return "", err
+	}
+	if svc.Spec.Type == corev1.ServiceTypeLoadBalancer {
+		return svc.Spec.LoadBalancerIP, nil
```

Comment:
I have seen most cloud providers use `svc.Status.LoadBalancer.Ingress.Hostname` or `svc.Status.LoadBalancer.Ingress.IP` and don't use the `spec.LoadBalancerIP`.
Let's first either of the above (if present) in this order - hostname -> ip -> spec.loadBalancerIP
If all are empty then return spec.ClusterIP itself.